include/cose/cose.h

```diff
@@ -205,6 +210,15 @@ typedef enum {
 	COSE_Curve_Ed448 = 7,
 } COSE_Curves;
 
+#if defined(COSE_C_USE_MBEDTLS)
+typedef struct mbedtls_ecp_keypair eckey_t;
+#else
+typedef struct eckey_t {
+	struct ec_key_st* key;
+	int group;
+} eckey_t;
+#endif	// COSE_C_USE_MBEDTLS
+
```

Comment:
As noted in the other pull request, this is not the way that I think this should go in the end.  It would be better to declare a COSE_Key structure and have a method to build that.  This can this be changed internal to the library without issues as well as adding some other types of keys.  If you are doing this, please just make this specific to MBEDTLS as passing in the structure is not how one would do this for OpenSSL, instead you would pass in an EVP_KEY *.  

I don't know what the current support is in MBEDTLS for TEMs at the moment.  I would expect that at some point in the future this is going to be added if not currently present and that would require yet a different possible internal change for that support.